pandas/tests/indexes/test_setops.py

```diff
@@ -463,3 +466,33 @@ def test_setop_with_categorical(index, sort, method):
     result = getattr(index, method)(other[:5], sort=sort)
     expected = getattr(index, method)(index[:5], sort=sort)
     tm.assert_index_equal(result, expected)
+
+
+@pytest.mark.parametrize("index_maker", tm.index_subclass_makers_generator())
```

Comment:
use the index fixture instead

Reply:
How should I handle cases where that fixture gives inappropriate input? 

Should I define a separate fixture for this? Can I do something like a [`hypothesis.assume`](https://hypothesis.readthedocs.io/en/latest/details.html#making-assumptions) to make sure some of the cases don't get through and aren't marked as skipped?

The cases which won't work here are at least:

* `"empty"`
* `"repeats"`

Reply:
its kind of ugly, but in many of these tests we do something like

```
def test_foo(index_fixture):
    if whatever(index_fixture):
        pytest.skip()
```

Reply:
(id be psyched to see a hypothesis-based implementation of some of our tests, need a way to generate any-valid-index)

Reply:
I've switched to the fixture, but it adds a lot of skipped tests. I would be nice if I could filter this before collection, or use a fixture where I can specify some features of the construction (e.g. values are unique and it has a length of `n`)